### DISCUSSION TITLE
stitched maps and selecting specific grains

### DISCUSSION THREAD

**Discussion by amirtopaz** [mtex-toolbox/mtex#602] [Ask Anything]
Hi, I have two questions:
1. I have HKL Channel 5 stitched file of multiple sites, but for some reason plenty of the data is missing. see attached for example
2. is there a way to calculate grains from a double-type array?
for example, the following code:
[omega,a,b] = fitEllipse(grains);
Deg_omega = omega/degree;
Deg_omega = Deg_omega(Deg_omega<10 & Deg_omega>-10);

now after the manipulations I want to go back to having this array as grains and calculate aspect ratio and etc..
Thanks for the help,
Amir

![orientation map](https://user-images.githubusercontent.com/62837798/100535165-0e496c80-321f-11eb-8cfc-a2944d5b5ed1.jpg)

**Comment by kilir** [comment]
Hi
I guess your question related to 1) is "Where is the data?" ? A guess would be that the stitched maps have different phases and one or some phases are not specified correctly during import. Compare what you have in your stitched maps and in the individual maps when loading them into mtex.

Wrt 2: You can do
```
selected_grains = grains(Deg_omega<10 & Deg_omega>-10);
```

Hope that helps.

Cheers,
Rüdiger

**Reply by amirtopaz** [reply]
I tried it, and I get an error:

Error using any
Dimension argument must be a positive integer scalar, a vector of unique positive integers, or 'all'.

Error in subsind (line 61)
    sub = any(subs{i}, find(size(subs{i}')==max(size(ind)),1));

Error in grain2d/subsref (line 22)
    ind = subsind(grains,s(1).subs);

Error in MTEX_loadEBSD_Plag_Amph (line 389)
selected_grains = grains(Deg_omega<10 & Deg_omega>-10);

**Reply by amirtopaz** [reply]
and regarding the stitched map.. when the maps are loaded individually, they are fine. the problem is only with the stitched map that simply skips on most of the data

**Reply by kilir** [reply]
With respect to your grain selection, the following works for me:
```
omega = fitEllipse(grains);
plot(grains(omega<10*degree | omega>170*degree));
```
Cheers,
Rüdiger

**Comment by ralfHielscher** [comment]
Could you please post the output of
```
ebsd
```
********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
ralf.hielscher@mathematik.tu-chemnitz.de
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Reply by amirtopaz** [reply]
Is that what you meant?


ebsd = EBSD  
 
 Phase  Orientations     Mineral        Color  Symmetry  Crystal reference frame
     0  203655 (46%)  notIndexed                                                
     1  207772 (47%)  notIndexed                                                
     2  21292 (4.8%)  Hornblende  light green     12/m1        X||a*, Y||b, Z||c
     3   835 (0.19%)   Anorthite   light blue        -1              X||a*, Z||c
     4   5981 (1.4%)  Quartz-new    Goldenrod       321        X||a*, Y||b, Z||c
     5  1616 (0.37%)    Titanite   LightCoral     12/m1        X||a*, Y||b, Z||c
     6   134 (0.03%)      Rutile     DarkBlue     4/mmm                         
     7   695 (0.16%)     Epidote    DarkGreen     12/m1        X||a*, Y||b, Z||c
 
 Properties: bands, bc, bs, error, mad, reliabilityindex, x, y
 Scan unit : um
 cprInfo: show struct

**Reply by kilir** [reply]
I remember that some software did strange things to the phase numbers when stitching maps which had different phases indexed (it's been in the old forum but a few years back, but can't find it right now). It seems you are using an Oxford system, right? a) If you load a .ctf, could you share the header of the file? b) you could try if exporting to another format might help (.h5oina)?

Cheers,
Rüdiger
